tests/subsys/settings/fcb/src/settings_test_fcb.c

```diff
@@ -122,10 +122,10 @@ int c1_handle_export(void (*cb)(char *name, char *value),
 	}
 
 	settings_str_from_value(SETTINGS_INT8, &val8, value, sizeof(value));
-	cb("myfoo/mybar", value);
+	ARG_UNUSED(cb("myfoo/mybar", value));
```

Comment:
This look like slight misuse of the macro. To my understanding it's intended to say "this argument is intentionally not used in this function", whereas in your case you want to say that the return value of the cb function is intentionally ignored (if I understand this right).  A more common way of handling this (at least what I've seen elsewhere) is a simple void cast in front of the function call, accompanied by an optional preceding code comment explaining why it's safe to ignore the return value.